src/test/test-proc-cmdline.c

```diff
@@ -247,6 +248,21 @@ TEST(proc_cmdline_key_startswith) {
         assert_se(!proc_cmdline_key_startswith("foo-bar", "foo_xx"));
 }
 
+static void test_proc_cmdline_filter_pid1_args_one(const char *s, const char *expected) {
+        _cleanup_free_ char *result;
+
+        assert_se(proc_cmdline_filter_pid1_args(s, &result) >= 0);
+        assert_se(streq(result, expected));
+}
+
+TEST(proc_cmdline_filter_pid1_args) {
+        test_proc_cmdline_filter_pid1_args_one("   systemd   ", "");
+        test_proc_cmdline_filter_pid1_args_one("systemd hoge  -x foo \t   --aaa var", "hoge foo var");
+        test_proc_cmdline_filter_pid1_args_one("systemd -a systemd.log-level=debug -b -c --quu --hoge 3", "systemd.log-level=debug 3");
+        test_proc_cmdline_filter_pid1_args_one("/usr/lib/systemd/systemd --switched-root  --system  --deserialize   30 systemd.log_level=debug --unit foo.target systemd.log_target=\"console\"  'arg   with   space '   \t",
+                                               "systemd.log_level=debug systemd.log_target=\"console\" 'arg   with   space '");
+}
```

Comment:
maybe a simpler, safer fix would be to change things so that we generate `--deserialize=30` instead of `--deserialize 30`? i kinda prefer the `=` syntax anyway, and it removes any ambiguity around this.

Reply:
Yeah, we shoud do that too, but as a separate PR. We want the fix in this PR anyway, because people can add arguments on their own, or upgrade via daemon-reexec from old systemds.

Reply:
> maybe a simpler, safer fix would be to change things so that we generate `--deserialize=30` instead of `--deserialize 30`? i kinda prefer the `=` syntax anyway, and it removes any ambiguity around this.

Now the suggestion is also addressed in this PR.